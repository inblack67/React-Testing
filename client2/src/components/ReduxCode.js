import React from 'react'
import { connect } from 'react-redux'
import { increment, decrement } from '../actions/count';

const ReduxCode = ({ countState: { count }, increment, decrement }) => {

    const onIncrement = e => {
        increment();
    }

    const onDecrement = e => {
        decrement();
    }

    return (
        <div>
            <span data-testid='count'>{count}</span>
            <a href="#!" onClick={onIncrement}>Increment</a>
            <a href="#!" onClick={onDecrement}>Decrement</a>
        </div>
    )
}

const mapStateToProps = state => ({
    countState: state.CountState
})

export default connect(mapStateToProps, { increment, decrement })(ReduxCode);

