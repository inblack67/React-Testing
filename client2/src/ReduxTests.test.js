import React from 'react';
import { fireEvent, render, cleanup } from '@testing-library/react';
import ReduxCode from './components/ReduxCode';
import { Provider } from 'react-redux'
import store from './store'

afterEach(cleanup);

function renderWithRedux(component){
    return {
        ...render(<Provider store={store}>
            {component}
        </Provider>)
    }
}

it('renders with redux', () => {
    const { asFragment, getByTestId } = render(<Provider store={store}>
        <ReduxCode />
    </Provider>);

    expect(asFragment).toMatchSnapshot();

    expect(getByTestId('count')).toHaveTextContent('0');

});

it('can increment on click', () => {
    // const { getByTestId, getByText } = render(<Provider store={store}>
    //     <ReduxCode />
    // </Provider>);

    // fireEvent.click(getByText('Increment'));

    // expect(getByTestId('count')).toHaveTextContent('1');

});

it('can decrement on click', () => {
    const { getByTestId, getByText } = renderWithRedux(<ReduxCode />);

    fireEvent.click(getByText('Decrement'));

    expect(getByTestId('count')).toHaveTextContent('-1');

});