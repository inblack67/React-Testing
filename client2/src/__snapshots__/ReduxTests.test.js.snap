// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with redux 1`] = `[Function]`;
